### ISSUE TITLE
more nameref assignment shenanigans

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#1544]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

_ekvstore_nonatomic_rm () {
  declare -n ns2="$1"
  local key="$2"

  unset ns2["$key"]
}
```

#### Here's what shellcheck currently says:
```
Line 4	SC2034: ns2 appears unused. Verify use (or export if used externally).
```
#### Here's what I wanted or expected to see:

To be consistent with #1543, the error should be at line 7.

**Comment by pabs3** [comment]
This appears to be a duplicate of #817.

**Comment by matthewpersico** [comment]
See also #1543
